x-pack/test/apm_api_integration/common/registry.ts

```diff
@@ -155,7 +153,7 @@ export function RegistryProvider({ getService }: FtrProviderContext) {
                 await supertest
                   .get('/api/ml/saved_objects/sync')
                   .set('kbn-xsrf', 'foo')
-                  .auth(ApmUsername.editorUser, APM_TEST_PASSWORD);
+                  .auth(ApmUsername.editorUser, kbnTestConfig.getUrlParts().password);
```

Comment:
I think that `kbnTestConfig.getUrlParts().password` will not always work here. It provides the password for whatever user is used for running Kibana (in most cases this will be the `elastic` user - but doesn't always have to be). The tests are passing locally / on PR CI because there the `elastic` user happens to have the same password (`changeme`), but if you e.g. point these tests at an ESS cluster where `elastic` is using a different password, I'm sure this bit won't work anymore.

Reply:
@pheyos But in that case the existing situation where the password is hardcoded to `APM_TEST_PASSWORD` should have also failed ?

Reply:
I think hardcoded passwords work as long as you can make sure that you've setup the corresponding users with that password during test runtime - I guess that's the case for `ApmUsername.editorUser`. But hardcoded passwords don't work for the `elastic` user. And obviously it also doesn't work in environments where you can't create users (like serverless).

Reply:
I see your point here. So these tests are for Stateful Kibana.
As you already found, we are manually creating the users here with different roles.

Regarding running the stateful tests, pointing to ESS, i believe this can be done by setting the `TEST_KIBANA_PASSWORD` and thats exactly what we use as password while creating our users. 

For serverless, i understand. I guess its a topic for separate discussion

Reply:
@achyutjhunjhunwala My recommendation is to only use `kbnTestConfig.getUrlParts().password` for the `elastic` user (or even better for the `kbnTestConfig.getUrlParts().username` - which almost always will be `elastic`) and not use it for users that are generated during test runtime - these should go with static passwords that are known in the test, like your `APM_TEST_PASSWORD`.  I'd also be careful in situations where you have e.g. `username: ApmUsername | 'elastic';` and use the same password for both as this might break in some environments.
Having said that, it's ultimately your decision as you're owning these tests, I'm just pointing out the risks :wink:

Reply:
@achyutjhunjhunwala I think this is good advice from @pheyos . Perhaps we should either use 
```
const { username, password } = kbnTestConfig.getUrlParts();
```

or credentials we control (eg `"apmMonitorClusterAndIndices"` + `APM_TEST_PASSWORD`) but not a mix

Reply:
In that case for serverless we can switch the username and password from the testConfig directly for now as we are not creating users for Serverless (must also fix the `username` typing)